web/packages/teleport/src/DeviceTrust/DeviceTrustLocked.tsx

```diff
@@ -0,0 +1,178 @@
+/**
+ * Copyright 2023 Gravitational, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import styled, { useTheme } from 'styled-components';
+
+import { Box, Text, Flex, Link } from 'design';
+
+import { IconCircle } from 'design/Icon/IconCircle';
+
+import { Windows, Linux, Apple } from 'design/Icon';
+
+import { LockIcon } from 'design/SVGIcon';
+
+import Table, { Cell } from 'design/DataTable';
+
+import {
+  DeviceListProps,
+  TrustedDeviceOSType,
+} from 'teleport/DeviceTrust/types';
+
+import {
+  FeatureBox,
+  FeatureHeader,
+  FeatureHeaderTitle,
+} from 'teleport/components/Layout';
+import { CtaEvent } from 'teleport/services/userEvent';
+import { ButtonLockedFeature } from 'teleport/components/ButtonLockedFeature';
+
+export function DeviceTrustLocked() {
+  const theme = useTheme();
+
+  function generateFakeItems(count) {
+    const items = [];
+    const osType = ['Windows', 'Linux', 'macOS'];
+
+    for (let i = 0; i < count; i++) {
+      items.push({
+        id: `id-${i}`,
+        assetTag: `CLFBDAACC${i}`,
+        enrollStatus: `enroll-status-${i}`,
+        osType: osType[i % osType.length],
+      });
+    }
+
+    return items;
+  }
```

Comment:
very small nit: this could be moved outside of the function so it doesn't rerun every render... although we really only render this once so it doesn't matter.

Reply:
moved outside in https://github.com/gravitational/teleport/pull/30249/commits/b5078ca2d60839ffef0921e717518035e74e26ca so it does not breaks the recommended pattern